x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils/import_rules_utils.ts

```diff
@@ -0,0 +1,396 @@
+                  });
+                } else if (rule != null) {
+                  resolve(
+                    createBulkErrorObject({
+                      ruleId,
+                      statusCode: 409,
+                      message: `rule_id: "${ruleId}" already exists`,
+                    })
+                  );
+                }
+              } catch (err) {
+                resolve(
+                  createBulkErrorObject({
+                    ruleId,
+                    statusCode: err.statusCode ?? 400,
+                    message: err.message,
+                  })
+                );
+              }
+            } catch (error) {
+              reject(error);
+            }
+          });
+          return [...accum, importsWorkerPromise];
+        }, [])
+      );
+      importRuleResponse = [...importRuleResponse, ...newImportRuleResponse];
+    }
+
+    return importRuleResponse;
+  }
+};
+
+// TODO: Batch this upfront and send down to check against
+export const checkExceptions = async ({
+  ruleId,
+  exceptions,
+  exceptionsClient,
+}: {
+  ruleId: string;
+  exceptions: ListArray;
+  exceptionsClient: ExceptionListClient | undefined;
+}): Promise<[BulkError[], ListArray]> => {
+  let ruleExceptions: ListArray = [];
+  let errors: BulkError[] = [];
+
+  if (!exceptions.length || exceptionsClient == null) {
+    return [[], exceptions];
+  }
+  for await (const exception of exceptions) {
+    const { list_id: listId, namespace_type: namespaceType } = exception;
+    const list = await exceptionsClient.getExceptionList({
+      id: undefined,
+      listId,
+      namespaceType,
+    });
+
+    if (list != null) {
+      ruleExceptions = [...ruleExceptions, { ...exception, id: list.id }];
+    } else {
+      // if exception is not found remove link
+      errors = [
+        ...errors,
+        createBulkErrorObject({
+          ruleId,
+          statusCode: 400,
+          message: `Rule with rule_id: "${ruleId}" references a non existent exception list of list_id: "${listId}". Reference has been removed.`,
+        }),
+      ];
+    }
+  }
+
+  return [errors, ruleExceptions];
+};
+
+export const importExceptionsHelper = async ({
+  exceptions,
+  exceptionsClient,
+  overwrite,
+  maxExceptionsImportSize,
+}: {
+  exceptions: Array<ImportExceptionsListSchema | ImportExceptionListItemSchema>;
+  exceptionsClient: ExceptionListClient | undefined;
+  overwrite: boolean;
+  maxExceptionsImportSize: number;
+}) => {
+  if (exceptionsClient == null) {
+    return {
+      success: true,
+      errors: [],
+      successCount: 0,
+    };
+  }
+
+  try {
+    const {
+      errors,
+      success,
+      success_count: successCount,
+    } = await exceptionsClient.importExceptionListAndItemsAsArray({
+      exceptionsToImport: exceptions,
+      overwrite,
+      maxExceptionsImportSize,
+    });
+
+    return {
+      errors,
+      success,
+      successCount,
+    };
+  } catch (err) {
+    throw new Error(err);
```

Comment:
This doesn't do anything right? You're just catching the error and then re-throwing it? No logging, or anything like that. You could just not wrap it and let it bubble up if it catches?

Reply:
Good catch! I'd wrapped a bunch of these during debugging sessions to console/locate the source of error. Removed!